Runtime/UI/Utils/SpriteStretch.cs

```diff
@@ -23,25 +21,17 @@ public class SpriteStretch : MonoBehaviour
         /// </summary>
         [SerializeField, Range(0f,1f)] private float spriteWidthOffsetPercentage;
         
-        private float worldSpaceWidth;
-        private float worldSpaceHeight;
-        
         private void Start()
         {
-            if (spriteRenderer.sprite != null && stretchOnStart)
+            if (spriteRenderer.sprite != null && stretchOnStartUsingMainCamera)
             {
-                CalculateCameraSize();
-                StretchSprite();
+                var cameraSize = GetCameraSize(Camera.main);
+                StretchSprite(cameraSize);
             }
         }
 
-        public void StretchSprite()
+        public void StretchSprite(Vector2 spaceSize)
```

Comment:
What is the spaceSize? Might need a more descriptive name, please provide context first to decide